packaging/material/packaging_python.md

```diff
@@ -0,0 +1,180 @@
+---
+type: slide
+slideOptions:
+  transition: slide
+  width: 1400
+  height: 900
+  margin: 0.1
+---
+
+<style>
+  .reveal strong {
+  font-weight: bold;
+    color: orange;
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+</style>
+
+# How to prepare a Python code for packaging
+
+---
+
+## What is PyPI?
+
+- [PyPI](https://pypi.org/) = **The Python Package Index** is a repository of software developed in the Python community
+- PyPI itself is developed on GitHub through another software called [Warehouse](https://github.com/pypa/warehouse)
+- PyPI has a informative [public dashboard](https://p.datadoghq.com/sb/7dc8b3250-85dcf667bd?from_ts=1636972832484&to_ts=1636976432484&live=true) to show its activity
+- Who maintains PyPI and pip? A big community! For example check out [Donald Stufft](https://twitter.com/dstufft/)
+- A major advantage is the active maintenance of PyPI, which is fueled by a big community. PyPI is also on [Twitter](https://twitter.com/PyPI)
+
+---
+
+## Example of PyPI package: fenicsprecice
+
+- Having a look at [fenicsprecice](https://pypi.org/project/fenicsprecice/)
+
+---
+
+## What is pip?
+
+- pip is sometimes referred to as a *package installer* and sometimes as a *package management system*
+- For all practical purposes it is an installer for packages available at PyPI
+- pip is itself a [package](https://pypi.org/project/pip/) which is available on PyPI
+- pip is open-source and is developed on [GitHub](https://github.com/pypa/pip)
+- pip has many more things which are worth exploring! [Look here!](https://pip.pypa.io/en/latest/development/)
+
+---
+
+## Python Enhancement Proposals (PEPs)
+
+- PEP is an evolving deisgn document which provides information regarding new features of Python, new processes and new environments
+- PEPs typically involve concise technical information which also acts as standardizations
+- Packaging workflows are also standardized in through multiple PEPs. Examples are:
+    - [PEP 427](https://www.python.org/dev/peps/pep-0427/) introduces the built-package format "wheel"
+- It is always a good idea to look for **PEP** conventions before trying out processes
+
+---
+
+## Questions?
+
+---
+
+## File structure for code packaging 1/2
+
+- Most essential component of packaging is file structure. Basic file structure is as follows:
+
+package_name/
+└── src/
+    └── solver/
```

Comment:
Why solver? Could a term more generic than "solver" be used? `executable`? Then it is immediately clear what this file does. "solver" (might) be somewhat ambiguous depending on the background of the students.

Reply:
I say solver because in the exercises the students get a file `solver.py` and the general nomenclature dictates that the directory name be the same as the core functionality name